### PR TITLE
Handle hashtags in spoiler_texts (partial fix for #699)

### PR DESCRIPTION
**Pull request by rtucker**
These commits add partial support for hashtags in the spoiler_text (aka content warning header).

The first commit covers parsing of spoiler_texts for hashtags, allowing them to be found by hashtag search.  The second commit creates links for hashtags in spoiler_texts on the server-generated views.

An attempt was made to link spoiler_text hashtags from the main web UI, but JavaScript is not my forte.  I believe `app/assets/javascripts/components/components/status_content.jsx` is where that change would need to be made, however.

### CODE REVIEW

File: app/services/process_hashtags_service.rb [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 class ProcessHashtagsService < BaseService
   def call(status, tags = [])
-    tags = status.text.scan(Tag::HASHTAG_RE).map(&:first) if status.local?
+    text = status.text
+    text << " " + status.spoiler_text unless status.spoiler_text.blank?
```
Comment (Gargron) [inline app/services/process_hashtags_service.rb]: This might fail due to immutable strings..?

Comment (rtucker) [inline app/services/process_hashtags_service.rb]: Plausible.  Is also relatively ugly due to rusty Ruby skills.  d6f97e0 may be better?

File: app/services/process_hashtags_service.rb [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
 class ProcessHashtagsService < BaseService
   def call(status, tags = [])
-    text = status.text
-    text << " " + status.spoiler_text unless status.spoiler_text.blank?
+    text = [status.text, status.spoiler_text].reject(&:empty?).join(' ')
```
Comment (beatrix-bitrot) [inline app/services/process_hashtags_service.rb]: this seems nicer yes

**Review by beatrix-bitrot — APPROVED** [review]
only issue from the last review was corrected so marking this approved

**Review by beatrix-bitrot — DISMISSED** [review]
ack, i spoke too soon, this has conflicts that need to be resolved!

**Review by rtucker — COMMENTED** [review]
Whoops!  I've fixed up the conflict in 9130073 but github does not seem to be recognizing it as such.  Hrmmph.

**Comment by rtucker** [comment]
I have no idea how to indicate that the requested changes have been made, but they have been (by resolving the conflict and updating the branch).

**Review by Gargron — APPROVED** [review]

**Pull request merged**